packages/modules/token/src/utils/verifier/index.ts

```diff
@@ -0,0 +1,47 @@
+import { TokenRegistry } from "@/src/type";
+import { InfinitCallback } from "@infinit-xyz/core";
+import { ContractInfo, verifyContract } from "@infinit-xyz/core/internal";
+import { Etherscan } from "@nomicfoundation/hardhat-verify/src/internal/etherscan.js";
+import { getArtifacts } from "@utils/artifact";
+import { Address, Chain, createPublicClient, http, PublicClient } from "viem";
+
+type BlockExplorerParams = {
+    apiKey?: string
+    apiUrl: string
+    url: string
+}
+
+export class TokenVerifier {
+  etherscan: Etherscan
+  client: PublicClient
+    constructor(chain: Chain, rpcEndpoint: string, params: BlockExplorerParams) {
+        params.apiKey = params.apiKey ?? "API_KEY_NOT_PROVIDED" // note: test that we can use default value
```

Comment:
why do we need to assgin `API_KEY_NOT_PROVIDED` to `params.apiKey`, can't we just assign `""`